### PR TITLE
Week 4 - Test services with not global Context

### PR DESCRIPTION
**Pull request by Tynukua**

### CODE REVIEW

File: src/main/scala/karazin/scala/users/group/week4/homework/services.scala [hunk 1]
```diff
@@ -18,39 +19,40 @@ import scala.util.Success
 import scala.util.Failure
 
 import karazin.scala.users.group.week4.homework.model._
+import scala.concurrent.ExecutionContext
+// import concurrent.ExecutionContext.Implicits.global
```
Comment (IgorWolkov) [inline src/main/scala/karazin/scala/users/group/week4/homework/services.scala]: Remove it please.

File: src/test/scala/karazin/scala/users/group/week4/homework/ProgramSuite.scala [hunk 2]
```diff
@@ -13,10 +19,23 @@ import scala.concurrent.Future
...
+        ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor)
     }
+    override def afterAll(): Unit = {
+      ctx.shutdown()
+    }
+  }
+  override def munitFixtures = List(context)
+  test("successful async test example") {
+      val views = getPostsViews()(using context())
+      views.map(v => print(v))(using executor = context())
```
Comment (IgorWolkov) [inline src/test/scala/karazin/scala/users/group/week4/homework/ProgramSuite.scala]: No `print(v)`! Remove it please. Everywhere.

**Review by IgorWolkov — CHANGES_REQUESTED** [review]
Some minor comments.

File: src/test/scala/karazin/scala/users/group/week4/homework/ProgramSuite.scala [hunk 3]
```diff
@@ -13,10 +19,22 @@ import scala.concurrent.Future
...
+      ctx = 
+        ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor)
     }
+    override def afterAll(): Unit = {
+      ctx.shutdown()
+    }
+  }
+  override def munitFixtures = List(context)
+  test("should get posts using ctx from hooks") {
+      val views = getPostsViews()(using context())
```
Comment (IgorWolkov) [inline src/test/scala/karazin/scala/users/group/week4/homework/ProgramSuite.scala:39]: Looks like this test checks nothing.

File: src/test/scala/karazin/scala/users/group/week4/homework/ServicesSuite.scala [hunk 4]
```diff
@@ -12,13 +17,22 @@ import scala.concurrent.Future
...
+      ctx = 
+        ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor)
+    }
+    override def afterAll(): Unit = {
+      ctx.shutdown()
     }
   }
+  override def munitFixtures = List(context)
+  test("should get profile using ctx from hooks") {
+      val profile = getUserProfile(using context())
```
Comment (IgorWolkov) [inline src/test/scala/karazin/scala/users/group/week4/homework/ServicesSuite.scala:35]: Looks like this test checks nothing.

**Review by IgorWolkov — CHANGES_REQUESTED** [review]
The tests should be improved.

File: src/test/scala/karazin/scala/users/group/week4/homework/ProgramSuite.scala (hunk 3 above)
Comment (Tynukua) [reply src/test/scala/karazin/scala/users/group/week4/homework/ProgramSuite.scala:39]: it tests context creating and using 

check some data is impossible cuz it's random
checking the type of ret value is also meanless

**Review by IgorWolkov — COMMENTED** [review]
The solution needed to be discussed in person.

**Review by IgorWolkov — APPROVED** [review]
Let it be.

**Pull request merged**